lib/scss-parser.es6

```diff
@@ -9,6 +9,50 @@ export default class ScssParser extends Parser {
         this.tokens = scssTokenizer(this.input);
     }
 
+    rule(tokens) {
+        let colonIndex = null;
+        let isSpaceAfterColon = false;
+        let value = null;
+
+        if (tokens[1] && tokens[1][0] === ':') {
+            colonIndex = 1;
+        } else if (tokens[2] && tokens[2][0] === ':') {
+            colonIndex = 2;
+        }
+        if (colonIndex) {
+            isSpaceAfterColon = tokens[colonIndex + 1] &&
+                tokens[colonIndex + 1][0] === 'space';
+            value = isSpaceAfterColon ?
+                tokens[colonIndex + 2] && tokens[colonIndex + 2][1] :
+                tokens[colonIndex + 1] && tokens[colonIndex + 1][1];
+        }
+
+        // If the conditions for compiling it as a nested prop (and not
+        // a selector as with `a :before`) are right
+        if (colonIndex &&
+            (value === null || isSpaceAfterColon || /[0-9.$]/.test(value[0]))
+        ) {
+            let decl;
+            tokens.pop();
+            super.decl(tokens);
+            decl = this.current.nodes[this.current.nodes.length - 1];
+            decl.nested = true;
+            decl.nodes = [];
+            // So that it'd work as a PostCSS container
+            Object.defineProperty(decl, 'push', { enumerable: false,
```

Comment:
This and the previous statements are actually hacks - for the parser to correctly work with Declaration instance as a container (like [here](https://github.com/postcss/postcss/blob/master/lib/parser.es6#L338)). It would be easier to be able to instantiate the declaration with [the new class](https://github.com/dryoma/postcss-scss/blob/nested-props/lib/declaration-nested.es6) from the beginning.